### DISCUSSION TITLE
Why does the tracking script need to be named `plausible.js` ?

### DISCUSSION THREAD

**Discussion by sunneydev** [plausible/analytics#2123] [Self-Hosted Support] [answered]
Adblockers filter JS scripts based on filenames. Why is there no way to change the script name or let just it be generated for each website add?

**Comment by metmarkosaric** [comment, marked as answer]
Hi @sunneydev! The name we recommend is script.js rather than plausible.js but adblockers also work based on domain names. If you're looking for more accurate data, do take a look [at our proxy](https://plausible.io/docs/proxy/introduction). There are several ways to set up a proxy all depending on what technology you use for your site. The one that's very easy to start with for anyone is [our Cloudflare proxy](https://plausible.io/docs/proxy/guides/cloudflare). It takes a couple of minutes to setup with our simple step by step instructions.

**Reply by sunneydev** [reply]
Yeah, but I don't think it's possible to just choose a name of the script file when self-hosting? And since I'm hosting Plausible it should be on my own domain. So it makes no sense for me to just use a proxy for the script file. 

I tried forking, editing the source and building my own docker image, but I came across some errors after running the server.

Could you point me to the right direction? How can I accomplish this?

**Reply by sunneydev** [reply]
@metmarkosaric Basically I'm asking if there's a way to change the script name without necessarily interfering with the source code? It's not mentioned in the docs.

**Reply by sunneydev** [reply]
Ok, I just noticed that `script.js` is available by default. My bad.
